src/app/api/mux/route.ts

```diff
@@ -0,0 +1,58 @@
+import Mux, { type Asset } from "@mux/mux-node";
+import { eq } from "drizzle-orm";
+import { headers } from "next/headers";
+import type { NextRequest } from "next/server";
+
+import { env } from "~/env.mjs";
+import { db } from "~/server/db";
+import { videoEntries } from "~/server/db/schema";
+
+type MuxWebhookAssetBody = {
+  type: string;
+  object: {
+    type: "asset";
+    id: string;
+  };
+  data: Asset;
+};
+
+export async function POST(req: NextRequest) {
+  try {
+    const signature = headers().get("mux-signature");
+
+    if (!signature) {
+      return new Response("Missing signature", { status: 400 });
+    }
+
+    const body = (await req.json()) as MuxWebhookAssetBody;
```

Comment:
You should instead create a zod schema that will `safeParse` the body.

Reply:
Schema added 👍